### ISSUE TITLE
ld architecture errors running rake on OSX 10.6.7

### ISSUE THREAD

**Issue by pmuellr** [smparkes/zxing.rb#1]
When trying to run rake to build zxing.rb, I was getting the following errors from the cc -o zxing.bundle command:

```
ld: warning: ignoring file zxing.o, file was built for unsupported file format which is not the architecture being linked (i386)
ld: warning: ignoring file /Users/pmuellr/Projects/zxing.rb/vendor/zxing/cpp/build/libzxing.a, file was built for archive which is not the architecture being linked (i386)
```

The final g++ invocation and cc invocations looked like this:

```
g++ -I. -I/System/Library/Frameworks/Ruby.framework/Versions/1.8/usr/lib/ruby/1.8/universal-darwin10.0 -I/System/Library/Frameworks/Ruby.framework/Versions/1.8/usr/lib/ruby/1.8/universal-darwin10.0 -I. -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE   -fno-common -I/Users/pmuellr/Projects/zxing.rb/vendor/zxing/cpp/core/src   -c zxing.cc

cc -arch i386 -arch x86_64 -pipe -bundle -undefined dynamic_lookup -o zxing.bundle zxing.o -L. -L/System/Library/Frameworks/Ruby.framework/Versions/1.8/usr/lib -L. -arch i386 -arch x86_64  -lstdc++ /Users/pmuellr/Projects/zxing.rb/vendor/zxing/cpp/build/libzxing.a -liconv   -lruby  -lpthread -ldl  
```

I tried running the cc command, by hand, removing all the -arch options except for a single -arch x86_64, and it linked with no warnings.

I was able to get this to work without changing any source by invoking rake a little differently:

```
ARCHFLAGS=”-arch x86_64” rake
```

There may still be problems; I'm still building RMagick :-) ; but I thought I'd mention it before I forgot.

**Comment by smparkes** [comment]
Thanks. Vaguely remember something like that. Honestly, I haven't tried to build zxing.rb since Lion came out, either ...

**Comment by pmuellr** [comment]
After building RMagick, was able to run rake zxing:test to completion.

Recommend just adding a note about the "arch" thing to your README.

**Comment by smparkes** [comment]
Cool. If you want, feel free to fork and send a pull request (then you credit).

**Comment by smparkes** [comment]
Fixed in db672e68949508e473bcdec45aa5e06eb8205d29
